src/fastapibootcamp/handlers/astroplan/endpoints.py

```diff
@@ -0,0 +1,148 @@
+"""Path operations for the astroplan router."""
+
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Path, Query
+from safir.models import ErrorLocation
+
+from fastapibootcamp.dependencies.requestcontext import (
+    RequestContext,
+    context_dependency,
+)
+from fastapibootcamp.exceptions import ObserverNotFoundError
+
+from .models import (
+    ObservabilityResponseModel,
+    ObservationRequestModel,
+    ObserverModel,
+)
+
+astroplan_router = APIRouter(tags=["astroplan"])
+
+# The core of a RESTful API is the "resource". In this API, observing sites are
+# the resource. This first endpoint lets the user retrieve an existing
+# observing site resource by its ID.
+#
+# e.g. GET /astroplan/observers/rubin
+
+# We can declare path parameters that are used commonly across multiple
+# endpoints in a single place.
+
+observer_id_type = Annotated[
```

Comment:
Minor nit: I think the convention for type aliases is to use camel-case (so `ObserverId`). Also, because this is a type alias, it's now recommended that you write this as `ObserverId: TypeAlias = ...` for various reasons explained here:

https://mypy.readthedocs.io/en/stable/kinds_of_types.html#type-aliases

Reply:
Thinking about this further, I'd probably name this `ObserverIdPathParam`, since it's not the type of a generic observer ID but very specifically the type of a FastAPI path parameter for an observer ID.

Reply:
Thanks. This is much better! And then thanks to ruff I learned about the new `type` keyword for Python 3.12. https://docs.astral.sh/ruff/rules/non-pep695-type-alias/

Reply:
Oh but mypy doesn't support it!